physicsnemo/utils/generative/sfm_samplers.py

```diff
@@ -0,0 +1,231 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Callable
+from typing import Dict
+
+import nvtx
+import torch
+import torch.nn as nn
+from omegaconf import DictConfig
+
+from physicsnemo.models.diffusion import SongUNetPosEmbd
+
+
+def sigma(t):
+    return t
+
+
+def sigma_inv(sigma):
+    return sigma
+
+
+@nvtx.annotate(message="SFM_encoder_sampler", color="red")
+def SFM_encoder_sampler(
+    networks: Dict[str, torch.nn.Module],
```

Comment:
Similar comment as for `SFMPrecondEmpty`. I think this can either be stashed in a `utils` file local to the example and not in the `physicsnemo` core, or alternately support an option in the `SF_sampler` to allow running "encoder-only" sampling.